### PR TITLE
Add JSON schema validation in Repair

### PR DESCRIPTION
**Pull request by adrien-berchet**
Also fix broken URLs and add Py310 and Py311

### CODE REVIEW

**Comment by mgeplf** [comment]
can you have a look at this @eleftherioszisis? thanks.

File: neuror/axon.py [hunk 1]
```diff
@@ -50,6 +50,7 @@ def _tree_distance(sec1, sec2):
         sec2 = sec2.parent
         dist += 2
         if None in {sec1, sec2}:
+            # pylint: disable=broad-exception-raised
```
Comment (eleftherioszisis) [inline neuror/axon.py:53]: I would suggest adding a general NeuroRError and raise that.

File: neuror/main.py [hunk 2]
```diff
@@ -277,6 +327,7 @@ def __init__(self,  # pylint: disable=too-many-arguments
         self.donated_intact_axon_sections = []
         self.repair_flags = repair_flags or {}
         self.params = params if params is not None else _PARAMS
+        self.validate_params(self.params)
```
[the remark this answers is not in the document]
Comment (adrien-berchet) [reply neuror/main.py:330]: Well, if the values are not wrong it should not break :innocent: 
But yeah I can make it optional for safety.

**Comment by adrien-berchet** [comment]
Thanks!

**Pull request merged**